### PR TITLE
Refactor Extension Landing Pages

### PR DESCRIPTION
**Pull request by technoboy10**
### Resolves:
This PR resolves issue #1982 

### Changes:

This PR refactors the extension landing pages to make it easier to create new landing pages without reusing as much code. This was done primarily by creating a number of reusable components. The components added were:
* **ExtensionLanding**: Class that extension landing pages extend; sets up state needed for every extension page (OS chooser, mainly)
  * **ExtensionSection**: Wrapper for sections on the extension landing page, handles some layout
  * **ExtensionHeader**: Component that wraps extension information (is passed an image URL for the right side of the header)
  * **ExtensionRequirements**: Display component that lays out extension requirements provided to it
  * **InstallScratchLink**: Component that renders Scratch Link installation instructions based on the provided OS
  * **ProjectCard**
  * **TipBox**
* **Steps**: Component to lay out numbered steps
  * **Step**: Component that takes a number and renders its children underneath or next to a circle with the provided number in it

### Possible issues
I wasn't sure if I should put the Steps/Step component inside the extension-landing folder or not, but decided to put it as its own component because it could potentially be reused in a lot of different places. This may have not been the right call, but it can be changed pretty easily.

### CODE REVIEW

File: src/views/ev3/ev3.jsx [hunk 1]
```diff
@@ -3,8 +3,8 @@
  * into something that can be used in both places (scratch-www#1982)
  */
 
-const bindAll = require('lodash.bindall');
 const injectIntl = require('react-intl').injectIntl;
```
Comment (chrisgarrity) [inline src/views/ev3/ev3.jsx]: Nit: remove the `TODO` comment saying we need to refactor the code :)

File: src/views/ev3/ev3.jsx [hunk 2]
```diff
@@ -13,328 +13,201 @@ const Page = require('../../components/page/www/page.jsx');
...
-                                        </span>
-                                        <div className="step-image">
-                                            <img src="/images/ev3/ev3-motor-port-a.png" />
-                                        </div>
-                                    </FlexRow>
-                                </FlexRow>
+                </ExtensionSection>
+                <ExtensionSection className="blue things-to-try">
+                    <h2><FormattedMessage id="ev3.thingsToTry" /></h2>
+                    <h3>Make a motor move</h3>
```
Comment (chrisgarrity) [inline src/views/ev3/ev3.jsx]: The text here should be FormattedMessage strings

File: src/views/ev3/ev3.jsx [hunk 3]
```diff
@@ -13,328 +13,201 @@ const Page = require('../../components/page/www/page.jsx');
...
-                </div>
-                <div className="faq inner">
+                        </Step>
+                    </Steps>
+                    <hr />
+                    <h3>Starter Projects</h3>
+                    <Steps>
+                        <ProjectCard
+                            cardUrl="https://downloads.scratch.mit.edu/ev3/ev3-wave-hello.sb3"
+                            description="Make a puppet robot and have a friendly chat."
```
Comment (chrisgarrity) [inline src/views/ev3/ev3.jsx]: The project cards should also be translatable.

File: src/views/ev3/ev3.scss [hunk 4]
```diff
@@ -1,313 +1,8 @@
...
-            background-image: url("/svgs/extensions/download.svg");
-            background-repeat: no-repeat;
-            width: 20px;
-            height: 20px;
-            vertical-align: text-top;
-            content: "";
-        }
-    }
-
     .extension-header {
```
Comment (chrisgarrity) [inline src/views/ev3/ev3.scss]: It's beautiful how simple this css gets!

File: src/views/microbit/microbit.jsx [hunk 5]
```diff
@@ -3,8 +3,8 @@
  * into something that can be used in both places (scratch-www#1982)
  */
```
Comment (chrisgarrity) [inline src/views/microbit/microbit.jsx]: Nit: remove TODO comment

File: src/views/ev3/ev3.jsx (hunk 3 above)
Comment (technoboy10) [reply src/views/ev3/ev3.jsx]: The l10n issues are being tracked in #1996 - should they be fixed in this PR or in a separate PR for that issue?

Comment (chrisgarrity) [reply src/views/ev3/ev3.jsx]: Sorry - I didn't see that one. It should be a separate PR!

**Pull request merged**